src/python_testing/TC_CADMIN_1_19.py

```diff
@@ -0,0 +1,173 @@
+            TestStep(4, "TH_CR1 reads the SupportedFabrics attribute from the Node Operational Credentials cluster. Save max_fabrics"),
+            TestStep(5, "Repeat the following steps (5a and 5b) max_fabrics - initial_number_of_fabrics times"),
+            TestStep(
+                "5a", "TH_CR1 send an OpenCommissioningWindow command to DUT_CE using a commissioning timeout of max_window_duration", "{resDutSuccess}"),
+            TestStep("5b", "TH creates a controller on a new fabric and commissions DUT_CE using that controller",
+                     "Commissioning is successful"),
+            TestStep(6, "TH reads the CommissionedFabrics attributes from the Node Operational Credentials cluster.",
+                     "Verify this is equal to max_fabrics"),
+            TestStep(
+                7, "TH_CR1 send an OpenCommissioningWindow command to DUT_CE using a commissioning timeout of max_window_duration", "{resDutSuccess}"),
+            TestStep(8, "TH creates a controller on a new fabric and commissions DUT_CE using that controller",
+                     "Verify DUT_CE responds with NOCResponse with a StatusCode field value of TableFull(5)"),
+            TestStep(9, "Repeat the following steps (9a and 9b) for each controller (TH_CRn) created by this test"),
+            TestStep("9a", "The controller reads the CurrentFabricIndex from the Node Operational Credentials cluster. Save as fabric_index."),
+            TestStep("9b", "TH_CR1 sends the RemoveFabric command to DUT_CE", "{resDutSuccess}"),
+            TestStep(10, "TH reads the CommissionedFabrics attributes from the Node Operational Credentials cluster.",
+                     "Verify this is equal to initial_number_of_fabrics."),
+        ]
+
+    def pics_TC_CADMIN_1_19(self) -> list[str]:
+        return ["CADMIN.S"]
+
+    @async_test_body
+    async def test_TC_CADMIN_1_19(self):
+        self.step(1)
+
+        # Establishing TH1 and TH2
+        self.th1 = self.default_controller
+
+        self.step(2)
+        GC_cluster = Clusters.GeneralCommissioning
+        attribute = GC_cluster.Attributes.BasicCommissioningInfo
+        duration = await self.read_single_attribute_check_success(endpoint=0, cluster=GC_cluster, attribute=attribute)
+        self.max_window_duration = duration.maxCumulativeFailsafeSeconds
+
+        self.step(3)
+        fabrics = await self.get_fabrics(th=self.th1)
+        initial_number_of_fabrics = len(fabrics)
+
+        self.step(4)
+        OC_cluster = Clusters.OperationalCredentials
+        max_fabrics = await self.read_single_attribute_check_success(dev_ctrl=self.th1, fabric_filtered=False, endpoint=0, cluster=OC_cluster, attribute=OC_cluster.Attributes.SupportedFabrics)
+
+        self.step(5)
+        fids_ca_dir = {}
+        fids_fa_dir = {}
+        fids = {}
+        self.print_step("total fabrics", max_fabrics - initial_number_of_fabrics)
+        for fid in range(1, max_fabrics - initial_number_of_fabrics):
```

Comment:
Shouldn't range start at 0? If `max_fabrics - initial_number_of_fabrics` is 1 you will currently loop through this 0 times which seems incorrect

Reply:
You are correct, when I made a prior change this appears to have gotten changed by accident.
I have updated this to start at int 0 rather than start at int 1.